File: app/src/main/java/com/example/resourcesapp/ui/coding/CodingViewModel.java
package com.example.resourcesapp.ui.coding;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class CodingViewModel extends ViewModel {

    private final MutableLiveData<String> mText;

    public CodingViewModel() {
        mText = new MutableLiveData<>();
        mText.setValue("This is slideshow fragment");
    }

    public LiveData<String> getText() {
        return mText;
    }
}

File: app/src/main/java/com/example/resourcesapp/ui/coding/CodingViewModel.kt
package com.example.resourcesapp.ui.coding

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class CodingViewModel : ViewModel() {
    private val mText: MutableLiveData<String>

    init {
        mText = MutableLiveData()
        mText.value = "This is slideshow fragment"
    }

    val text: LiveData<String>
        get() = mText
}